draft-thomson-http-oblivious.md

```diff
@@ -925,6 +925,152 @@ information in {{media-types}} for the media types "message/ohttp-req",
 
 --- back
 
+
+# Complete Example of a Request and Response
+
+<!-- Generated using ohttp:
+RUST_LOG=ohttp cargo test -p ohttp -\-lib -\- -\-nocapture request_response
+-->
+
+A single request and response exchange is shown here. Binary values (key
+configuration, secret keys, the content of messages, and intermediate values)
+are shown in hexadecimal. The request and response here are absolutely minimal;
+the purpose of this example is to show the cryptographic operations.
+
+The oblivious request resource generates a key pair. In this example the server
+chooses the DHKEM(X25519, HKDF-SHA256) and generates an X25519 key pair
+{{?X25519=RFC7748}}. The X25519 secret key is:
+
+~~~
+15ce887006e079dcc7d67e73e5c13e31a55083f816eca9ebcf523b90ea2ab7b0
+~~~
+
+The oblivious request resource constructs a key configuration that includes the
+corresponding public key as follows:
+
+~~~
+0100200020f21c612398e4384c21b7f2a759133c6c2d1b9ce6d033613dfad2c7
+3d4826214900080001000100010003
+~~~
+
+This key configuration is somehow obtained by the client. Then when a client
+wishes to send an HTTP request of a GET request to `https://example.com`, it constructs the following binary HTTP message:
+
+~~~
+00034745540568747470730b6578616d706c652e636f6d012f
+~~~
+
+The client then reads the oblivious request resource key configuration, selects a pair of KDF and AEAD and generates an X25519 key pair:
```

Comment:
```suggestion
The client then reads the oblivious request resource key configuration and selects a mutually supported KDF and AEAD. In this example, the client selects HKDF-SHA256 and AES-128-GCM. The client then generates an X25519 key pair:
```

Reply:
I'm not really sure how I feel about this particular part, since it exposes HPKE internals. Maybe we could rewrite this to something like so:

```
The client then generates an HPKE context using the server's public key, yielding an encapsulated 
key:

<X25519 public key>

With the corresponding private key:

<X25519 private key>
```

Reply:
Agree. Will fix.